test/component/Legend.spec.tsx

```diff
@@ -712,26 +712,29 @@ describe('<Legend />', () => {
       test.each(expectedLegendTypeSymbolsWithColor('#3182bd'))(
         'should render element $selector for legendType $legendType',
         ({ legendType, selector, expectedAttributes }) => {
-          const { container, debug } = render(
+          const { container } = render(
             <LineChart width={500} height={500} data={categoricalData}>
               <Legend />
               <Line dataKey="value" legendType={legendType} />
             </LineChart>,
           );
-          const [legendItem] = assertHasLegend(container);
-          const symbol = legendItem.querySelector(selector);
-          if (symbol == null) {
-            debug();
-          }
-          expect(symbol).toBeInTheDocument();
-          const expectedAttributeNames = Object.keys(expectedAttributes);
-          expect.soft(symbol?.getAttributeNames().sort()).toEqual(expectedAttributeNames.sort());
-          expectedAttributeNames.forEach(attributeName => {
-            expect.soft(symbol).toHaveAttribute(attributeName, expectedAttributes[attributeName]);
-          });
+          assertExpectedAttributes(container, selector, expectedAttributes);
         },
       );
     });
+
+    it('should prefer Legend.iconType over Line.legendType', () => {
+      const { container } = render(
+        <LineChart width={500} height={500} data={numericalData}>
+          <Legend iconType="circle" />
```

Comment:
The old code, inside `getLegendProps`, has `type: legendItem.props.iconType || item.props.legendType,`. I forgot to include that in my context :/ fixed now